### ISSUE TITLE
[Python client] Unable to serialize/deserialize complex schema

### ISSUE THREAD

**Issue by HugoPelletier** [apache/pulsar#11536]
**Describe the bug**

The solution push in the PR https://github.com/apache/pulsar/pull/11400 is working with a simple structure.
As soon as you have a more complex, including _Array_ type of field, a `ValueError` exception is raised

```python
        class NestedObj1(Record):
            na1 = String()
            nb1 = Double()
            nc1 = NestedObj0()

        class NestedObj2(Record):
            na2 = Integer()
            nb2 = Boolean()
            nc2 = NestedObj1()

        class ComplexRecord(Record):
            a = Integer()
            b = Integer()
            c = EnumObj
            nested = Array(NestedObj2())
            d = Array(String())
```

**Expected behavior**
We should be able to serialize/deserialize complex schema

**Comment by HugoPelletier** [comment]
FYI, @gaoran10, I already have a solution for this issue.

**Comment by HugoPelletier** [comment]
Closing this issue. Was fixed in #11530
